FrontEnd/src/components/admin/CreateUserForm.js

```diff
@@ -0,0 +1,134 @@
+import React, {useState} from "react";
+import {connect} from "react-redux";
+import generalStyles from "./../../styles/GeneralStyles"
+import {Typography} from "@material-ui/core";
+import Paper from "@material-ui/core/Paper";
+import TextField from "@material-ui/core/TextField";
+import IconButton from "@material-ui/core/IconButton";
+import NoteAddIcon from "@material-ui/icons/NoteAdd";
+import {SERVER} from "../../../index";
+import { USER_ADMINISTRATION_IDENTIFIER} from "../general/identifiers";
+import {switchContent} from "../../actions/teacherNavigationActions";
+import makeStyles from "@material-ui/core/styles/makeStyles";
+import FormControl from "@material-ui/core/FormControl";
+import InputLabel from "@material-ui/core/InputLabel";
+import Select from "@material-ui/core/Select";
+import MenuItem from "@material-ui/core/MenuItem";
+
+const customStyles = makeStyles((theme) => ({
+    backgroundPaper: {
+        width: '100%',
+        maxWidth: '800px',
+        display: 'flex',
+        flexDirection: 'column',
+        padding: theme.spacing(2),
+        justifyContent: "center",
+    },
+    formTextfield: {
+        marginTop: '2%',
+    },
+    formControl: {
+        marginTop: '2%',
+    },
+}))
+
+function CreateUserForm(props) {
+    const generalStyle = generalStyles();
+    const customStyle = customStyles();
+    const [firstname, setFirstname] = useState("");
+    const [lastname, setLastname] = useState("");
+    const [role, setRole] = useState("student");
+    const [firstnameFieldFilled, setFirstnameFieldFilled] =useState(true);
+    const [lastnameFieldFilled, setLastnameFieldFilled] =useState(true);
+
+    function handleFormChange(event, type) {
+        switch (type) {
+            case "firstname": setFirstname(event.target.value); break;
+            case "lastname": setLastname(event.target.value); break;
+            case "role": setRole(event.target.value);
+        }
+    }
+
+    function submitTaskForm(){
+        let isFirstnameFieldFilled = (firstname !== "");
+        let isLastnameFieldFilled = (lastname !== "");
+        setFirstnameFieldFilled(isFirstnameFieldFilled);
+        setLastnameFieldFilled(isLastnameFieldFilled);
```

Comment:
Probier mal setFirstnameFieldFilled( firstname => firstname !== "" ) 

Übrigens kannst du dank JS magic auch statt `firstname !== ""` `!firstname` benutzen

Reply:
auch wenn der Kommentar als Outdated markiert ist, ist das ja immernoch aktuell.
Zu 1. Ich weiß jetzt nicht genau was damit gemeint ist. Soll der ausdruck in der Klammer durch  [firstname !== ""]  ersetzt werden oder durch [firstname => firstname !== ""]  ?
Zum ersten Fall: da ich diesen Ausdruck ja zweimal brauche, werte ich ihn lieber einmal aus und speichere ihn in einer Variable, anstatt ihn zweimal auszuwerten. Kommt am ende aber auch aufs gleiche hinaus denk ich.
Im zweiten Fall: wenn ich das so schreibe, funktioniert die Abfrage nicht mehr.

zu 2. der JS Magic: gut zu wissen. Jedoch find ich persönlich die erste Variante lesbarer.
Ich denk mir immer so: ich schreib lieber ein bisschen mehr, wenn der Code dafür am ende lesbarer ist.

Reply:
zu 1. ja ich dachte an den zweiten Fall, schade dass des net funktioniert